check_api/src/main/java/com/google/errorprone/VisitorState.java

```diff
@@ -770,24 +771,20 @@ private static final class SharedState {
    * Like {@link Elements#getConstantExpression}, but doesn't over-escape single quotes in strings.
    */
   public String getConstantExpression(Object value) {
-    String escaped = getElements().getConstantExpression(value);
-    if (value instanceof String) {
-      // Don't escape single-quotes in string literals
-      StringBuilder sb = new StringBuilder();
-      for (int i = 0; i < escaped.length(); i++) {
-        char c = escaped.charAt(i);
-        if (c == '\\' && i + 1 < escaped.length()) {
-          char next = escaped.charAt(++i);
-          if (next != '\'') {
-            sb.append(c);
-          }
-          sb.append(next);
-        } else {
-          sb.append(c);
-        }
+    if (!(value instanceof CharSequence str)) {
```

Comment:
Unfortunately we're not _quite_ ready to start using Java > 11 features in non-test code internally. I can remove the `instanceof` pattern when importing this, I'm just leaving a note here in case you wonder why it disappears.